### PR TITLE
Adding a vsts-ci file

### PR DESCRIPTION
**Pull request by livarcocc**
Adding a vsts-ci file with our build definition so that we can set VS…TS for CI.

### CODE REVIEW

**Comment by mmitche** [comment]
@livarcocc For the eventual move to dotnet.visualstudio.com, this is not the form of YAML that we want.  We would like to use the arcade templates so that this definition works will before OSS as well as official, sends telemetry, etc.

**Comment by livarcocc** [comment]
@mmitche we will do this in steps. I talked to @markwilkie and we will move to arcade later on.

**Comment by livarcocc** [comment]
@chcosta

File: .vsts-ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        - agent.os -equals Windows_NT        
+      parallel: 2
+      matrix:
+        Build_Debug:
+          _BuildConfig: Debug
+          _PublishType: none
+          _SignType: test
+        Build_Release:
+          _BuildConfig: Release
+          ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
```
Comment (chcosta) [inline .vsts-ci.yml]: Did you intend to change the `Build.Reason` for all of this to just `PullRequest` in a separate PR when you do the official build changes?

File: .vsts-ci.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+- template: /eng/build.yml
+  parameters:
+    agentOs: Linux
+    dockerImage: microsoft/dotnet-buildtools-prereqs:ubuntu-14.04-cross-0cd4667-20170319080304
+    queue: 
+      ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+        name: DotNetCore-Linux
+      ${{ if notIn(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:        
+        name: DotNetCore-Test
+      demands:
```
Comment (chcosta) [inline .vsts-ci.yml]: indent this demands so it only applies to the `DotNetCore-Test` queue, shouldn't be needed for DotNetCore-Linux

File: .vsts-ci.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Build_Debug:
+          _BuildConfig: Debug
+          _PublishType: none
+          _SignType: test
+        Build_Release:
+          _BuildConfig: Release
+          _PublishType: none
+          ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+            _SignType: test
+            PB_PublishBlobFeedKey:
```
Comment (chcosta) [inline .vsts-ci.yml]: You could make `PB_PublishBobFeedKey` and `PB_PublishBlobFeedUrl` global variables.  ie. move these to the top of the file.

```YAML
variables:
  ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
    PB_PublishBlobFeedKey:
    PB_PublishBlobFeedUrl: 

phases:
- template: /eng/build.yml
…
```

File: eng/build.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                  --configuration $BuildConfig
+                  /p:PB_PublishBlobFeedKey=$BlobFeedKey
+                  /p:PB_PublishBlobFeedUrl=$BlobFeedUrl
+        displayName: Build
+        env:
+          BuildConfig: $(_BuildConfig)
+          BlobFeedKey: $(PB_PublishBlobFeedKey)
+          BlobFeedUrl: $(PB_PublishBlobFeedUrl)
+
+    - ${{ if notIn(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
```
Comment (chcosta) [inline eng/build.yml]: I don't think some of the below steps will actually work as-is.  Again, I presume you're ok with that and intend to fix them up when you do official build changes.  Depending on your timeline, you may just want to delete the internal build only sections for the sake of simplicity.  ie, remove the install MicroBuild plugin task, and the artifact publishing stuff.

Comment (chcosta) [reply eng/build.yml]: If you do remove the internal build stuff, then you might just simplify even further and collapse this whole thing into a single file (ie, move the "cibuild" stuff into the `.vsts-ci.yml` file.

File: .vsts-ci.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            PB_PublishBlobFeedUrl:
+          ${{ if notIn(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+            _PublishType: blob
+            _SignType: real
+
+
+- template: /eng/build.yml
+  parameters:
+    agentOs: Linux
```
Comment (chcosta) [inline .vsts-ci.yml]: Do you intend to add OSX later?

File: .vsts-ci.yml (hunk 1 above)
Comment (livarcocc) [reply .vsts-ci.yml]: Yes. Subsequent PR. Once I have pull request validation happening, I will switch to making CI working.

File: .vsts-ci.yml (hunk 2 above)
Comment (livarcocc) [reply .vsts-ci.yml]: Will do.

File: .vsts-ci.yml (hunk 3 above)
Comment (livarcocc) [reply .vsts-ci.yml]: Cool. Will do.

File: eng/build.yml (hunk 4 above)
Comment (livarcocc) [reply eng/build.yml]: I will keep them. I will work on getting official builds working today, as soon as this PR makes in.

File: .vsts-ci.yml (hunk 5 above)
Comment (livarcocc) [reply .vsts-ci.yml]: I was waiting for you to tell me the queue I should use for it 😄

Comment (chcosta) [reply .vsts-ci.yml]: Nice.  Yesterday I was told that the queue would be available today, but I'm not sure how confident that timeline is.  I have a meeting with Ilya in about 45 minutes and I'll ask how that work is progressing.  When it's populated, the queue will be "DotNetCore-Mac", but there's no machines there at the moment.

**Pull request merged**